dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/util/ContentUtils.java

```diff
@@ -475,53 +476,50 @@ public static long count(String query, User user, String tmDate) {
-				if(pullParents){
-					relNameForQuery = relationshipName.trim() + "-child";
-				}else{
-					relNameForQuery = relationshipName.trim() + "-parent";
-				}
-			}
 
 			return getPullResults(relationship, contentletIdentifier, condition, limit, -1, sort,
-					user, tmDate, relNameForQuery);
+					user, tmDate, pullParents);
 		}
 
-	/**
-	 * Logic used for `pullRelated` and `pullRelatedField` methods
-	 * @param relationship
-	 * @param contentletIdentifier
-	 * @param condition
-	 * @param limit
-	 * @param offset
-	 * @param sort
-	 * @param user
-	 * @param tmDate
-	 * @param relNameForQuery
-	 * @return
-	 */
-	private static List<Contentlet> getPullResults(final Relationship relationship,
-			String contentletIdentifier, final String condition, final int limit, final int offset, String sort, final User user,
-			final String tmDate, String relNameForQuery) {
-		if(!UtilMethods.isSet(relNameForQuery))//DOTCMS-5328
-			relNameForQuery = relationship.getRelationTypeValue();
+    /**
+     * Logic used for `pullRelated` and `pullRelatedField` methods
+     */
+    private static List<Contentlet> getPullResults(final Relationship relationship,
+            String contentletIdentifier, final String condition, final int limit, final int offset,
+            String sort, final User user,
+            final String tmDate, final boolean pullByParent) {
 
-		contentletIdentifier = RecurrenceUtil.getBaseEventIdentifier(contentletIdentifier);
+        contentletIdentifier = RecurrenceUtil.getBaseEventIdentifier(contentletIdentifier);
 
-		String pullquery = "+type:content +" + relNameForQuery + ":" + contentletIdentifier;
+        StringBuilder pullQuery = new StringBuilder();
+        try {
+            final Contentlet contentlet = conAPI
+                    .findContentletByIdentifierAnyLanguage(contentletIdentifier);
 
-		if(UtilMethods.isSet(condition)){
-			   pullquery += " " + condition;
-		}
+            final List<Contentlet> relatedContent = conAPI.getRelatedContent(contentlet, relationship, pullByParent, user, false);
+
+            if (relatedContent.isEmpty()){
+                return Collections.emptyList();
+            }
+
+            pullQuery.append("+identifier:(").append(String.join(" OR ", relatedContent
+                    .stream().map(cont -> cont.getIdentifier()).collect(
+                            Collectors.toList()))).append(")");
 
-		if(!UtilMethods.isSet(sort)){
-			sort = relationship.getRelationTypeValue() + "-" + contentletIdentifier + "-order";
-		}
-		return pull(pullquery, offset, limit, sort, user, tmDate);
-	}
 
-	/**
+            if (UtilMethods.isSet(condition)) {
+                pullQuery.append(" AND ").append(condition);
+            }
+
+            return pull(pullQuery.toString(), offset, limit, sort, user, tmDate);
+        } catch (Exception e) {
+            Logger.error(ContentUtils.class, "Error pulling related content", e);
```

Comment:
Please log the content's Identifier and relationship name too.

Reply:
Done